tests/partition_test.py

```diff
@@ -360,6 +361,14 @@ def test_cell_list_overflow(self):
     neighbors = neighbors.update(R)
     self.assertTrue(neighbors.did_buffer_overflow)
 
+    # Ensure that updating the neighbor matrix doesn't use i64 matrices
```

Comment:
Can you use `self.assertDtypesMatch` instead? Also, no need to print anything. 

https://github.com/google/jax/blob/2143f3d450a8a283b1327dac82826c7f8fe12d6e/jax/_src/test_util.py#L935

Also, would it make sense to just check for the dtype of `nbrs.idx` here and `cell_list.id_buffer.idx` in the cell list test above instead of inspecting the jaxpr? I think that typically the format of the jaxpr is not guaranteed to be stable over time so I could see this leading to regressions.

Reply:
Agreed, changed it to just check the return type. I thought there might have been some intermediate buffers that were i64 even if the return type is i32 but wasn't able to make that happen easily be removing some of the type annotations I added, so I think we're in good shape.